### PR TITLE
fix: user-id is clear through logout function

### PR DESCRIPTION
**Pull request by IamSudhir-Kumar**
## Related Issue
#175 

## Description

1. Modified `logoutUser` function in response to a potential issue.
2. Removed explicit session ID argument in the `deleteSession` method call.
3. Improved clarity and compatibility by using `deleteSession` without specifying a session ID.
4. Ensured a straightforward and effective logout process by clearing the current session.

### CODE REVIEW

File: src/backend/auth.api.ts [hunk 1]
```diff
@@ -190,7 +190,15 @@ const getCurrentUser = async () => {
  * @returns {Object} returns session
  */
 const logoutUser = async () => {
-  return await account.deleteSession("current");
+  try {
+    // Use the deleteSession method without specifying a session ID
+    const deletedSession = await account.deleteSession();
```
Comment (Sanchitbajaj02) [inline src/backend/auth.api.ts:195]: The session is clearing from the appwrite but not clearing from the cookies. Try to implement it in the redux where we are updating the cookies

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

File: src/redux/reducers/authReducer.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 "use client";
 import { createSlice } from "@reduxjs/toolkit";
 import type { PayloadAction } from "@reduxjs/toolkit";
-import { setCookie } from "nookies";
+import { setCookie } from "cookies";
```
Comment (Sanchitbajaj02) [inline src/redux/reducers/authReducer.ts:4]: you cannot directly access the cookies in nextjs client. That's why added the nookies package.

File: src/redux/reducers/authReducer.ts [hunk 3]
```diff
@@ -28,14 +28,24 @@ export const registerReducer = createSlice({
...
-      setCookie(null, "userId", "");
-      state.error = false;
-      state.creds.userId = "";
-      state.creds.email = "";
-      state.creds.createdAt = "";
-      state.creds.isVerified = false;
-      state.loading = false;
-    },
+    
+      try {
```
Comment (Sanchitbajaj02) [inline src/redux/reducers/authReducer.ts:32]: try catch idea is good. Is this tested?

File: src/redux/reducers/authReducer.ts [hunk 4]
```diff
@@ -28,14 +28,24 @@ export const registerReducer = createSlice({
...
+        // Clear user ID cookie
+        setCookie(null, "userId", "");
```
Comment (Sanchitbajaj02) [inline src/redux/reducers/authReducer.ts:34]: There is a function called destroy cookies in nookies. You should implement that.

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

**Comment by thevirengarg** [comment]
@IamSudhir-Kumar What's the progress on the changes requested?

**Comment by Girishbari** [comment]
@IamSudhir-Kumar #213 this was the changes

**Pull request closed without merging**